### PR TITLE
[circle-mpqsolver] Add Bisection Unitests

### PR DESCRIPTION
**Pull request by stamalakhov**
This draft refactors bisection to enable in-memory data input for Evaluator and BisectionSolver.

Previous draft: #11452
Related: #11374 
ONE-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

**Comment by stamalakhov** [comment]
@seanshpark @jinevening 
This draft redeclares some functions in `HDF5Importer` as `const`. Please let me know if this is prohibited.

File: compiler/circle-mpqsolver/src/bisection/BisectionSolver.h [hunk 1]
```diff
@@ -75,9 +81,11 @@ class BisectionSolver final : public MPQSolver
                  const std::string &def_quant, core::LayerParams &layers);
 
 private:
-  float _qerror = 0.f; // quantization error
+  float _qerror_ratio = 0.f; // quantization error ratio
```
Comment (jinevening) [inline compiler/circle-mpqsolver/src/bisection/BisectionSolver.h:84]: IIUC, this value has to be not changed after construction. Can you make it const?

**Comment by jinevening** [comment]
Overall looks good to me. The update in dio-hdf5 also looks good. Please split PR and upload :)

File: compiler/circle-mpqsolver/src/bisection/BisectionSolver.h (hunk 1 above)
Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/bisection/BisectionSolver.h:84]: > IIUC, this value has to be not changed after construction. Can you make it const?

ok.

**Pull request closed without merging**